### ISSUE TITLE
Vacuum: Status is always cleaning

### ISSUE THREAD

**Issue by MelleD** [rospogrigio/localtuya#1334]
I am currently trying to integrate my OKP Life with Tuya into Home Assistant.
However, I'm not sure which status DP and ID I have to use. Can anyone tell?
In addition, I also have problems selecting the DP for Fan Speed

Thanks in advance
<img width="408" alt="Bildschirmfoto 2023-03-16 um 23 21 42" src="https://user-images.githubusercontent.com/6340898/225767606-17ca82ab-e946-4465-a0d2-fadae419f3f4.png">

<img width="383" alt="Bildschirmfoto 2023-03-16 um 23 20 03" src="https://user-images.githubusercontent.com/6340898/225767502-112d3dd0-b7c7-40c4-90a3-71a3e4ff7289.png">

**Comment by MelleD** [comment]
Here is the json what i find in the iot tuya portal:
```
  "status": [
      {
        "code": "power",
        "dp_id": 1,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "power_go",
        "dp_id": 2,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "mode",
        "dp_id": 3,
        "type": "Enum",
        "values": "{\"range\":[\"standby\",\"smart\",\"chargego\",\"random\",\"wall_follow\",\"spiral\"]}"
      },
      {
        "code": "direction_control",
        "dp_id": 4,
        "type": "Enum",
        "values": "{\"range\":[\"forward\",\"backward\",\"turn_left\",\"turn_right\",\"stop\"]}"
      },
      {
        "code": "electricity_left",
        "dp_id": 6,
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      },
      {
        "code": "seek",
        "dp_id": 13,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "clean_record",
        "dp_id": 15,
        "type": "String",
        "values": "{\"maxlen\":255}"
      },
      {
        "code": "clean_area",
        "dp_id": 16,
        "type": "Integer",
        "values": "{\"unit\":\"㎡\",\"min\":0,\"max\":9999,\"scale\":0,\"step\":1}"
      },
      {
        "code": "clean_time",
        "dp_id": 17,
        "type": "Integer",
        "values": "{\"unit\":\"min\",\"min\":0,\"max\":9999,\"scale\":0,\"step\":1}"
      },
      {
        "code": "fault",
        "dp_id": 18,
        "type": "Bitmap",
        "values": "{\"label\":[\"edge_sweep\",\"middle_sweep\",\"left_wheel\",\"right_wheel\",\"garbage_box\",\"land_check\",\"collision\"]}"
      }
    ]
  },
```

**Comment by L4ngeFrank** [comment]
> Ok found the solution the data point id is mode = 3. There is no cleaning mode extra.
> 
> Then it works. For the locate i needed an own script to send 13 to true.

Hi, could ypu be more specific where and how to change this? I have the same issue in HA with the OKP KP2

![image](https://github.com/rospogrigio/localtuya/assets/37044307/ab8a8967-5e0b-4a76-b4bd-913803708c63)

**Comment by myromeo** [comment]
Hi, I too would like to understand how you solved this. I set mode to DP 3 but status is always cleaning. 

![Uploading IMG_9327.jpeg…]()

**Comment by MelleD** [comment]
Go to the following page https://auth.tuya.com/ and looked at the dp directly in the REST Api and looked for it and then simply entered it

> Hi, could you be more specific where and how to change this? I have the same issue in HA with the OKP KP2
> Thanks!

Not sure what you mean there are two "problems" described.

**Comment by myromeo** [comment]
@MelleD does your correctly update the state? Mines working as far as the integration goes but state is always ‘cleaning’. 

It seems the Tuya api doesn’t provide a state, only a cleaning mode. Unfortunately ‘standby’ is a cleaning mode. Perhaps it needs some magic to remap the cleaning mode to state. No idea how to do that however!

![IMG_9328](https://github.com/rospogrigio/localtuya/assets/60740643/4bce47e0-58a2-4d08-833c-a90735c4d4e6)

**Comment by MelleD** [comment]
Yes you cannot edit the mode you have to delete it and re add it. 
This mentioned in the docs:

I have mode list:
smart,wall_follow,spiral,single
docked status:
charging,chargecompleted

returning:
chargego

pause status:
paused

stop status:
standby

**Comment by myromeo** [comment]
I’ve made a small change and removed ‘standby’ from idle and added it to docked. Now it correctly tells me when it is docked as that seems to be the only time it uses standby status.

**Comment by L4ngeFrank** [comment]
Great guys, thanks for your support, i'm a little further now, but the status in HA still stays on 'cleaning" but the K2P is charging.
For the record, i'm using LocalTuya.

![image](https://github.com/rospogrigio/localtuya/assets/37044307/ebea413e-59a6-4bcf-a396-ae2e1b582e59)

**Comment by myromeo** [comment]
@L4ngeFrank delete the vacuum completely and re-add it in local Tuya. When you first set up the entities the first one to pick is ‘ID’, this should be DP 3 (same as mode). Everything else as per my previous post and it should work. It took me a bit to realise you have to change the ID entity to 3 for the status to work. You can only change this on initial setup.

**Comment by L4ngeFrank** [comment]
> @L4ngeFrank delete the vacuum completely and re-add it in local Tuya. When you first set up the entities the first one to pick is ‘ID’, this should be DP 3 (same as mode). Everything else as per my previous post and it should work. It took me a bit to realise you have to change the ID entity to 3 for the status to work. You can only change this on initial setup.

Thank U!!! That did the trick.

**Comment by jezzaaa** [comment]
Thanks for the suggestion, @myromeo. I set the ID to the Mode DP (27). Now my card now shows "idle" instead of "cleaning" when the vac is docked, so that's an improvement.

However, I believe it's supposed to show "docked". The Tuya integration shows the docked state as "docked" on the vacuum card, so I'd have thought it would show the same for LocalTuya.

The status also shows as "cleaning" when the vac is returning to charge. Tuya integration shows "Returning to dock", which makes more sense.

**Comment by chrisrosenkreuz23** [comment]
Ok guys, I fixed it. Myromeo had the right idea but it does not yield the wanted result. Here is what I did: the "ID" should NOT be same as mode. It should be the same as teh "status" code in the tuya platform. It is logical, since what we are looking for in HA says just that: Status. It is missing in the list above that the OP initially posted. Mine shows up as 28 in the tuya iot platform

        "code": "status",
        "custom_name": "",
        "dp_id": 28,
        "time": 1716006011829,
        "value": "charging"
        
thing is, changing it to this made the commands break even though it fixed the status. I then went back into the integration to edit the device config, and saw the power dp was changed. I then changed it to the value of "power_go" which was 25

        "code": "power_go",
        "custom_name": "",
        "dp_id": 25,
        "time": 1716006011834,
        "value": false
        
adunno if it differs from brand to brand but mine was something called v-tac5555. 
Now everything works as intended. Thanks all for contributing
